packages/babel-preset-mapbox/index.js

```diff
@@ -25,130 +42,151 @@ module.exports = function preset(context, opts) {
       '`@mapbox/babel-preset-mapbox` expects an object as an option.'
     );
   }
+
   // prevent ambiguity if user is using both
-  if (
-    (process.env.BROWSERSLIST && opts['babel-preset-env']) ||
-    (process.env.BROWSERSLIST && opts['env'])
-  ) {
+  if (process.env.BROWSERSLIST && opts['@babel/preset-env']) {
     throw new Error(
       'Please do not use process.env.BROWSERSLIST and `babel-preset-env` together.'
     );
   }
 
   opts = Object.assign({}, opts);
 
-  const isEnvDevelopment = env === 'development' || env === undefined;
-  const isEnvProduction = env === 'production';
-  const isEnvTest = env === 'test';
-  let payload = [
-    { name: 'babel-preset-react' },
+  const presets = [
+    {
+      name: '@babel/preset-react',
+      options: {
+        // Adds component stack to warning messages
+        // Adds __self attribute to JSX which React will use for some warnings
+        development: isEnvDevelopment || isEnvTest,
+        // Will use the native built-in instead of trying to polyfill
+        // behavior for any plugins that require one.
+        useBuiltIns: true
```

Comment:
Which specific JS features are required for the React preset's `useBuiltIns`? Let's note that.

Reply:
It isn't directly mentioned in the `preset-react` doc, but it is mentioned in https://babeljs.io/docs/en/next/babel-plugin-transform-react-jsx.html#usebuiltins that  `Object.assign` is the only feature. I kind of looked into the code too and I found a single [ref](https://github.com/babel/babel/blob/master/packages/babel-helper-builder-react-jsx/src/index.js#L198) to Object.assign.

Reply:
Yeah, let's add that to a comment 👍